### PR TITLE
feat: docgen CLI

### LINKED ISSUE #283: [Feature] Generate only markdown to plug into docute or Vuepress
Could we use VSG to generate just markdown files, that we can then plug into Vuepress or something like Docute or Docsify? From what I understood, it is not possible at the moment.

https://github.com/vue-styleguidist/vue-docgen-api/issues/38 might be connected to this, as such a CLI tool using the ouput from the API might be what I am asking for?

Thanks!

### PR DESCRIPTION
**Pull request by elevatebart**
cc @dobromir-hristov

closes #283 
- Commands
  - [x] `vue-docgen`
  - [x] `vue-docgen --watch`
- Config
  - [x] through command line
  - [x] custom templates
  - [x] configuration through file (easier than command line)
  - [x] module resolution config (for webpack alias and modules)
  - [x] load extra markdown per component 
       - [x] from external file
       - [x] from `<docs>` block
- Supported Items
  - [x] Props
  - [x] Methods
  - [x] Events
  - [x] Slots 
  - [x] Tags (use it to customize display)
- [x] documentation
- [ ] unit tests

### CODE REVIEW

**Comment by dobromir-hristov** [comment]
Looks good, a bit hard to follow in some places for me, but its because I am not familiar with the docgen api.
One thing that worries me is the `components` and `componentsRoot` configs. 

I guess the `getDocFileName` and `getDestFile` will get more documented?

I have not tested the outcome, will do that over the weekend.

**Comment by elevatebart** [comment]
Thank you @dobromir-hristov 
I added a lot of text in the readme to better explain the intended use.

I think I will deliver as is so that you guys can try it.
